core/primitives/src/types.rs

```diff
@@ -852,29 +847,17 @@ pub mod chunk_extra {
                     balance_burnt,
                     congestion_info: congestion_info.unwrap(),
                     bandwidth_requests: bandwidth_requests.unwrap(),
-                })
-            } else if ProtocolFeature::CongestionControl.enabled(protocol_version) {
```

Comment:
This is incorrect. Assume `ProtocolFeature::CongestionControl.enabled(protocol_version)` is true. We should be returning Self::V3(ChunkExtraV3...

Change parameter `congestion_info` type from `Option<CongestionInfo>` to `CongestionInfo`